app/components/UI/PermissionsSummary/PermissionsSummary.tsx

```diff
@@ -409,28 +425,130 @@ const PermissionsSummary = ({
     );
   }
 
+  const calculateBottomSheetHeight = useMemo(() => {
+    let currentBaseHeight = BOTTOM_SHEET_BASE_HEIGHT;
+    if (isNonDappNetworkSwitch) {
+      currentBaseHeight += 150;
+    }
+
+    if (accountAddresses.length <= 2) {
+      return currentBaseHeight;
+    }
+
+    const visibleItems =
+      accountAddresses.length >= MAX_VISIBLE_ITEMS
+        ? MAX_VISIBLE_ITEMS
+        : accountAddresses.length;
```

Comment:
```suggestion
    const visibleItems = Math.min(accountAddresses.length, MAX_VISIBLE_ITEMS);
```

maybe? idk. doesn't really matter